### PR TITLE
The ability to have all variables printed via a predicate

### PR DESCRIPTION
**Pull request by bbakerman**
This now uses the graphql-java implementation of DocumentCompiling and it also provides a hint allowing us to FF this into Nadel usage in the AGG

Please make sure you consider the following:

- [ x] Add tests that use __typename in queries
- [x ] Does this change work with all nadel transformations (rename, type rename, hydration, etc)? Add tests for this.
- [ / ] Is it worth using hints for this change in order to be able to enable a percentage rollout?
- [ / ] Do we need to add integration tests for this change in the graphql gateway?
- [ x ] Do we need a pollinator check for this?

### CODE REVIEW

File: engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt [hunk 1]
```diff
@@ -214,7 +216,12 @@ class NextgenEngine @JvmOverloads constructor(
             it.children.single()
         }
 
-        val (transformResult, executionPlan) = transformHydrationQuery(service, executionContext, actorField, serviceHydrationDetails)
+        val (transformResult, executionPlan) = transformHydrationQuery(
+            service,
+            executionContext,
+            actorField,
+            serviceHydrationDetails
+        )
```
Comment (bbakerman) [inline engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt:224]: Hmm IDEA seems to have done this formatting change.

File: test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt [hunk 2]
```diff
@@ -165,25 +167,35 @@ private suspend fun execute(
...
-                                    fail("Unable to match service call")
+                                    fail(
+                                        """Unable to match service call 
+                                        |   fixture : '${fixture.name}' 
+                                        |   service : '${serviceName}' 
+                                        |   query : '${actualQuery}' 
+                                        |   variables : '${actualVariables}' 
+                                        |   operation : '${actualOperationName}' 
+                                        """.trimMargin()
+                                    )
```
Comment (bbakerman) [inline test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt:198]: This makes the fail message a smidge easier to work out whats gone wrong.  I know its printed earlier but its now all near the last exception

File: test/src/test/resources/fixtures/document variable handling/inlined-all-arguments.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          "extensions": {}
+        }
+# language=JSON
+response: |-
+  {
+    "data": {
+      "hello": "world"
+    },
+    "extensions": {}
+  }
```
Comment (bbakerman) [inline test/src/test/resources/fixtures/document variable handling/inlined-all-arguments.yml:61]: This is the ALL as variables test case - the hook turns on the hint

The default is still JSON only

File: test/src/test/resources/fixtures/document variable handling/inlined-json-arguments.yml [hunk 4]
```diff
@@ -51,12 +51,12 @@ serviceCalls:
     - serviceName: MyService
       request:
         query: |
-          query myQuery($var_0: JSON, $var_1: JSON!) {
-            hello(arg: {payload : $var_0}, arg1: $var_1)
+          query myQuery($v0: JSON, $v1: JSON!) {
+            hello(arg: {payload : $v0}, arg1: $v1)
```
Comment (bbakerman) [inline test/src/test/resources/fixtures/document variable handling/inlined-json-arguments.yml:55]: naming of variables was shortened

File: engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/document/DocumentPredicates.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package graphql.nadel.enginekt.document
+
+import graphql.normalized.VariablePredicate
+
+class DocumentPredicates {
```
Comment (gnawf) [inline engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/document/DocumentPredicates.kt:5]: No need to fix, but just FYI you could've just made this an object i.e.

```
object DocumentPredicates {
  val jsonPredicate
  val allVariablesPredicate
}
```

And the behavior would be the same

File: test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt (hunk 2 above)
Comment (gnawf) [reply test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt:198]: Nice!

**Review by gnawf — APPROVED** [review]
Looks good

**Pull request merged**